### PR TITLE
block account creartion by mx record in addition to domain name

### PR DESCRIPTION
**Pull request by dylanwh**
This needs a bit of testing.

The AntiSpam stuff is on the admin.cgi paged (linked there).
Add one of the things returned by 'dig -t mx gmail.com' and then see if account creation is restricted for example mozilla.com. This should all work in the container.

(Until the new deps are built, it will fail because Net::DNS doesn't exist. But by tomorrow running bmo-refresh-bundle should correct that)

### CODE REVIEW

File: template/en/default/robots.txt.tmpl [hunk 1]
```diff
@@ -18,6 +18,8 @@ Disallow: [% basepath FILTER none %]show_bug.cgi*format=multiple*
 
 Allow: [% basepath FILTER none %]describecomponents.cgi
 Allow: [% basepath FILTER none %]describekeywords.cgi
+Disallow: [% basepath FILTER none %]showdependency*
+Disallow: [% basepath FILTER none %]banmyipforever.cgi
```
Comment (kyoshino) [inline template/en/default/robots.txt.tmpl]: This `robots.txt` change is a copy of #854, has to be removed.

**Comment by dylanwh** [comment]
yup, woops

**Comment by purelogiq** [comment]
@dylanwh I'm not able to verify that the intended behavior is working.

- Ran `bmo-refresh-bundle`, pulled this branch, and reprovisioned the vm.
- Confirmed that I can create a new account (at least to the point of it saying it sent an email confirmation) using a @gmail.com address.
- Confirmed that I cannot create a new account using one of the existing domains on the spam list.
```
(conduit) imadueme@conduitvm:~$ dig -t mx gmail.com

; <<>> DiG 9.10.3-P4-Ubuntu <<>> -t mx gmail.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 42385
;; flags: qr rd ra; QUERY: 1, ANSWER: 5, AUTHORITY: 0, ADDITIONAL: 10

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;gmail.com.			IN	MX

;; ANSWER SECTION:
gmail.com.		2440	IN	MX	20 alt2.gmail-smtp-in.l.google.com.
gmail.com.		2440	IN	MX	40 alt4.gmail-smtp-in.l.google.com.
gmail.com.		2440	IN	MX	5 gmail-smtp-in.l.google.com.
gmail.com.		2440	IN	MX	10 alt1.gmail-smtp-in.l.google.com.
gmail.com.		2440	IN	MX	30 alt3.gmail-smtp-in.l.google.com.

;; ADDITIONAL SECTION:
gmail-smtp-in.l.google.com. 94	IN	A	172.217.197.26
alt1.gmail-smtp-in.l.google.com. 223 IN	A	64.233.190.26
alt2.gmail-smtp-in.l.google.com. 191 IN	A	74.125.193.27
alt3.gmail-smtp-in.l.google.com. 95 IN	A	66.102.1.26
alt4.gmail-smtp-in.l.google.com. 95 IN	A	74.125.128.26
gmail-smtp-in.l.google.com. 94	IN	AAAA	2607:f8b0:400d:c0f::1b
alt1.gmail-smtp-in.l.google.com. 4 IN	AAAA	2800:3f0:4003:c01::1b
alt3.gmail-smtp-in.l.google.com. 95 IN	AAAA	2a00:1450:400c:c06::1b
alt4.gmail-smtp-in.l.google.com. 95 IN	AAAA	2a00:1450:4013:c02::1b

```
- I added `alt1.gmail-smtp-in.l.google.com` (tried with and without the `.` at the end) to the domain spamlist. (Let me know if I'm reading this MX record wrong).
- Was still able to create gmail accounts.

**Comment by dylanwh** [comment]
interesting! I wonder what I did different. I'll re-check and re-request review when I figure it out

**Comment by globau** [comment]
Removing stale PR with no associated bug.

**Pull request closed without merging**